src/utils_lgpl/ec_module/packages/ec_module/src/ec_netcdf_timeseries.f90

```diff
@@ -320,26 +333,24 @@ recursive function ecNetCDFScan(ncptr, quantity, location, q_id, l_id, dimids, v
       integer, dimension(:), allocatable :: dimids_check
 
       success = .false.
+      ltl = len_trim_nobnd(quantity)
 
       ! search for standard_name
       do ivar = 1, ncptr%nVars
-         ltl = len_trim(quantity)
-         if (strcmpi(trim(ncptr%standard_names(ivar)), trim(quantity))) exit
+         if (strcmpi(trim(ncptr%standard_names(ivar)), trim(quantity), ltl)) exit
```

Comment:
I would hope we can make this work instead using the existing `ncstdnames` approach, by having something like this in `ecProviderCreateNetCdfItems()`:
```
      case ('waterlevelbnd')
         ncvarnames(1) = 'waterlevel'
         ncstdnames(1) = 'sea_surface_height'
```
Let's refine this together under UNST-9376 for proper followup in next sprint.